ipp/src/parser.rs

```diff
@@ -120,7 +128,16 @@ impl ParserState {
             }
             if let Some(arr) = self.context.pop() {
                 if let Some(val_list) = self.context.last_mut() {
-                    val_list.push(IppValue::Collection(arr));
+                    let mut map: std::collections::HashMap<String, IppValue> = std::collections::HashMap::new();
+                    for idx in (0..arr.len()).step_by(2) {
+                        match (arr.get(idx), arr.get(idx + 1)) {
+                            (Some(IppValue::MemberAttrName(k)), Some(v)) => {
+                                map.insert(k.to_string(), v.clone());
+                            },
+                            _ => {}
```

Comment:
Are you sure? I was thinking about ignoring incorrect collection in case there is an error in printer.
You cannot fix the hardware and it can be irrelevant to the user.

Reply:
Right, let's keep it compatible with buggy firmwares.